FlowCrypt/Controllers/ImportKey/ImportKeyViewController.swift

```diff
@@ -0,0 +1,207 @@
+//
+//  ImportKeyViewController.swift
+//  FlowCrypt
+//
+//  Created by Anton Kharchevskyi on 14.11.2019.
+//  Copyright © 2019 FlowCrypt Limited. All rights reserved.
+//
+
+import AsyncDisplayKit
+import MobileCoreServices
+
+final class ImportKeyViewController: ASViewController<TableNode> {
+    private enum Parts: Int, CaseIterable {
+        case title, description, fileImport, pasteBoardImport
+
+        var indexPath: IndexPath {
+            IndexPath(row: self.rawValue, section: 0)
+        }
+    }
+
+    private let decorator: ImportKeyDecoratorType
+    private let pasteboard: UIPasteboard
+    private let dataManager: DataManagerType
+    private let core: Core
+
+    private var userInfoMessage = "" {
+        didSet { updateSubtitle() }
+    }
+
+    init(
+        decorator: ImportKeyDecoratorType = ImportKeyDecorator(),
+        pasteboard: UIPasteboard = UIPasteboard.general,
+        core: Core = Core.shared,
+        dataManager: DataManagerType = DataManager.shared
+    ) {
+        self.pasteboard = pasteboard
+        self.decorator = decorator
+        self.dataManager = dataManager
+        self.core = core
+        super.init(node: TableNode())
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupUI()
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        navigationController?.setNavigationBarHidden(false, animated: animated)
+        navigationController?.navigationBar.barStyle = .black
+    }
+
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        userInfoMessage = ""
+    }
+
+    private func setupUI() {
+        node.delegate = self
+        node.dataSource = self
+        title = decorator.sceneTitle
+    }
+
+    private func updateSubtitle() {
+        DispatchQueue.main.async {
+            self.node.reloadRows(at: [Parts.description.indexPath], with: .fade)
```

Comment:
Is it still necessary to push UI updates to main thread manually, even when using AsyncDisplayKit?

Reply:
https://github.com/FlowCrypt/flowcrypt-ios/issues/141